### PR TITLE
(Un)Subscribers not getting removed from the address book

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: Observer/Newsletter/ChangeContactSubscription.php [hunk 1]
```diff
@@ -103,17 +103,13 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
-                        \Dotdigitalgroup\Email\Model\Importer::IMPORT_TYPE_SUBSCRIBER_UPDATE,
-                        ['email' => $email, 'id' => $contactEmail->getId()],
-                        \Dotdigitalgroup\Email\Model\Importer::MODE_SUBSCRIBER_UPDATE,
-                        $websiteId
-                    );
-                }
+                //Add subscriber update to importer queue
+                $this->importerFactory->registerQueue(
+                    \Dotdigitalgroup\Email\Model\Importer::IMPORT_TYPE_SUBSCRIBER_UPDATE,
+                    ['email' => $email, 'id' => $contactEmail->getId()],
```
Comment (cdiacon) [inline Observer/Newsletter/ChangeContactSubscription.php]: @adeelq what is the $contactEmail->getId() coming from ???

Comment (adeelq) [inline Observer/Newsletter/ChangeContactSubscription.php]: @cdiacon https://github.com/adeelq/dotmailer-magento2-extension/blob/ee04ebfc52721358d1c24d63d02c72124ab58e36/Observer/Newsletter/ChangeContactSubscription.php#L80

Comment (cdiacon) [inline Observer/Newsletter/ChangeContactSubscription.php]: @adeelq there is no id field for contact table?!?

**Comment by adeelq** [comment]
@cdiacon if i'm not wrong  $contact->getId() is Identifier getter from Magento\Framework\Model\AbstractModel. Magento will always return the actual identifier defined in https://github.com/dotmailer/dotmailer-magento2-extension/blob/master/Model/ResourceModel/Contact.php#L12

**Pull request merged**